messaging/src/main/java/org/axonframework/eventhandling/MultiSourceTrackingToken.java

```diff
@@ -213,6 +219,10 @@ public boolean equals(Object o) {
 
         for (Map.Entry<String, TrackingToken> trackingTokenEntry : trackingTokens.entrySet()) {
             try {
+                if ( trackingTokenEntry.getValue() == null  && that.trackingTokens.get(trackingTokenEntry.getKey()) == null) {
```

Comment:
This can still lead to a NPE, when `trackingTokenEntry` is null, but `that` has a non-null entry.
Consider using `Object.equals(x, y)` in the already existing if statement, instead of `x.equals(y)`